src/sys/socket/addr.rs

```diff
@@ -502,7 +510,7 @@ impl SockAddr {
         match *self {
             SockAddr::Inet(InetAddr::V4(ref addr)) => (mem::transmute(addr), mem::size_of::<libc::sockaddr_in>() as libc::socklen_t),
             SockAddr::Inet(InetAddr::V6(ref addr)) => (mem::transmute(addr), mem::size_of::<libc::sockaddr_in6>() as libc::socklen_t),
-            SockAddr::Unix(UnixAddr(ref addr, len)) => (mem::transmute(addr), (len + mem::size_of::<libc::sa_family_t>()) as libc::socklen_t),
```

Comment:
The old calculation only worked on Linux.  On FreeBSD (and probably OSX, but I don't have an OSX machine to test on), bind()ing a SockAddr created this way would chop off the last character.